### ISSUE TITLE
Failing `unstable-cli` on deploying `istio`

### ISSUE THREAD

**Issue by khlifi411** [kyma-project/kyma#16982]
**Description**
The pipeline [unstable-cli](https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/unstable-cli) is failing consistently when deploying Kyma. Same as [nightly-main-reconciler-publish-unstable-cli]( https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/nightly-main-reconciler-publish-unstable-cli).

**Expected result**
Istio component deployment should succeed.

**Actual result**
Istio component deployment is failing.
```log
X Deploying Kyma

Deployed Components: 3/4
Could not deploy the following components:
- istio

Error: kyma deployment failed
```

**Failing Run link**:
https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/unstable-cli/1631556224048697344

**Comment by ruanxin** [comment]
same happens in [kyma-upgrade-k3d-kyma2-to-main](https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/kyma-upgrade-k3d-kyma2-to-main)

**Comment by triffer** [comment]
The latest reconciler version has extended error handling and will return additional errors. This means the latest reconciler won't work with Kyma version 2.11.2 unless you define a profile. 
To me it looks like this CI pipeline uses defaults that cause this issue. The values for defaults were adapted on Kyma main (https://github.com/kyma-project/kyma/pull/17028).

So imo the only option to fix this when using Kyma 2.11.2 is using a profile.

**Comment by ruanxin** [comment]
Hi @triffer,
This error also happens in [kyma-upgrade-k3d-kyma2-to-main](https://storage.googleapis.com/kyma-prow-logs/logs/kyma-upgrade-k3d-kyma2-to-main/1635189011280564224/build-log.txt) job, which will test install latest kyma release, then test kyma main.

And`Could not install Istio` also happens during deploy kyma 2.11.2
```
2023-03-13 08:02:07 UTC [INFO] PID:920 --- [main] install_kyma:118 ### Installing Kyma 2.11.2
- No previous Kyma version found
- Using Kyma from the workspace directory: /root/.kyma/sources/2.11.2
- Initial setup
Deploying Kyma
- Component 'CRDs' deployed
- Component 'cluster-essentials' deployed
- Component 'certificates' deployed
X Component 'istio' failed. Retrying...
Could not install Istio: Error occurred when calling istioctl: rendered IstioOperator yaml was: {"kind":"IstioOperator","apiVersion":"install.istio.io/v1alpha1","metadata":{"name":"default-operator","namespace":"istio-system","creationTimestamp":null},"spec":{"profile":"default","hub":"eu.gcr.io/kyma-project/external/istio","tag":"1.16.2-distroless","meshConfig":{"accessLogEncoding":"JSON","accessLogFile":"/dev/stdout","defaultConfig":{"holdApplicationUntilProxyStarts":true},"defaultProviders":{"tracing":[]},"enablePrometheusMerge":false,"enableTracing":true,"extensionProviders":[{"name":"kyma-traces","opencensus":{"context":["W3C_TRACE_CONTEXT"],"port":55678,"service":"telemetry-trace-collector-internal.kyma-system.svc.cluster.local"}},{"envoyFileAccessLog":{"logFormat":{"labels":{}},"path":"/dev/stdout"},"name":"envoy"},{"envoyFileAccessLog":{"logFormat":{"labels":{}},"path":"/dev/stdout"},"name":"stdout-json"}],"trustDomain":"cluster.local"},"components":{"pilot":{"enabled":true,"k8s":{"env":[{"name":"PILOT_HTTP10","value":"1"}],"podAnnotations":{"reconciler.kyma-project.io/managed-by-reconciler-disclaimer":"DO NOT EDIT - This resource is managed by Kyma.\nAny modifications are discarded and the resource is reverted to the original state.\n"},"resources":{"limits":{"cpu":"500m","memory":"1024Mi"},"requests":{"cpu":"100m","memory":"512Mi"}},"serviceAnnotations":{"reconciler.kyma-project.io/managed-by-reconciler-disclaimer":"DO NOT EDIT - This resource is managed by Kyma.\nAny modifications are discarded and the resource is reverted to the original state.\n"},"securityContext":{"runAsUser":"65534","runAsNonRoot":true,"runAsGroup":"65534","seccompProfile":{"type":"RuntimeDefault"}}}},"cni":{"enabled":true,"namespace":"istio-system","k8s":{"resources":{"limits":{"cpu":"500m","memory":"1024Mi"},"requests":{"cpu":"100m","memory":"512Mi"}}}},"ingressGateways":[{"enabled":true,"name":"istio-ingressgateway","k8s":{"hpaSpec":{"minReplicas":1,"maxReplicas":5,"metrics":[{"type":"Resource","resource":{"name":"cpu","targetAverageUtilization":80}},{"type":"Resource","resource":{"name":"memory","targetAverageUtilization":80}}]},"resources":{"limits":{"cpu":"2000m","memory":"1024Mi"},"requests":{"cpu":"100m","memory":"128Mi"}},"strategy":{"rollingUpdate":{"maxUnavailable":0,"maxSurge":"100%"}},"securityContext":{"runAsUser":"65534","runAsNonRoot":true,"runAsGroup":"65534","seccompProfile":{"type":"RuntimeDefault"}}}}],"egressGateways":[{"enabled":false,"name":"istio-egressgateway","k8s":{"resources":{"limits":{"cpu":"2000m","memory":"1024Mi"},"requests":{"cpu":"10m","memory":"120Mi"}}}}]},"values":{"cni":{"cniBinDir":"/bin","cniConfDir":"/var/lib/rancher/k3s/agent/etc/cni/net.d"},"gateways":{"istio-ingressgateway":{"autoscaleEnabled":true,"name":"istio-ingressgateway","podAnnotations":{"reconciler.kyma-project.io/managed-by-reconciler-disclaimer":"DO NOT EDIT - This resource is managed by Kyma.\nAny modifications are discarded and the resource is reverted to the original state.\n"},"serviceAnnotations":{"dns.gardener.cloud/class":"garden","dns.gardener.cloud/dnsnames":"*.local.kyma.dev","reconciler.kyma-project.io/managed-by-reconciler-disclaimer":"DO NOT EDIT - This resource is managed by Kyma.\nAny modifications are discarded and the resource is reverted to the original state.\n"}}},"global":{"imagePullPolicy":"IfNotPresent","priorityClassName":"kyma-system-priority","proxy":{"readinessFailureThreshold":40,"readinessInitialDelaySeconds":5,"readinessPeriodSeconds":5,"resources":{"limits":{"cpu":"250m","memory":"256Mi"},"requests":{"cpu":"10m","memory":"64Mi"}}},"proxy_init":{"resources":{"limits":{"cpu":"100m","memory":"50Mi"},"requests":{"cpu":"10m","memory":"10Mi"}}}},"pilot":null,"sidecarInjectorWebhook":{"enableNamespacesByDefault":false,"objectSelector":{"autoInject":true,"enabled":false},"rewriteAppHTTPProbe":true}}}} : All attempts fail:
#1: got error executing command /root/.kyma/istio/istio-1.16.2/bin/istioctl stderr: 
- Processing resources for Istio core.
✔ Istio core installed
- Processing resources for Istiod.2023-03-13T08:03:18.263090Z	error	installer	failed to update resource with server-side apply for obj HorizontalPodAutoscaler/istio-system/istiod: no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1"

```

**Comment by triffer** [comment]
This was fixed with version 2.11.3 and is included in 2.12.
